File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsOrgBrandingAdmin.kt
package com.lextures.android.core.lms

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement

@Serializable
data class OrgBrandingResponse(
    val logoUrl: String? = null,
    val faviconUrl: String? = null,
    val primaryColor: String = OrgBrandingAdminLogic.DEFAULT_PRIMARY_COLOR,
    val secondaryColor: String = OrgBrandingAdminLogic.DEFAULT_SECONDARY_COLOR,
    val customDomain: String? = null,
    val customEmailDisplayName: String? = null,
    val contrastWarningPrimary: Boolean? = null,
    val contrastRatioPrimary: Double? = null,
)

@Serializable
data class PutOrgBrandingRequest(
    val logoUrl: String? = null,
    val faviconUrl: String? = null,
    val primaryColor: String,
    val secondaryColor: String,
    val customDomain: String? = null,
    val customEmailDisplayName: String? = null,
)

@Serializable
data class OrgBrandingUploadResponse(
    val url: String? = null,
)

@Serializable
data class AiConfigResponse(
    val orgId: String? = null,
    val featuresEnabled: Map<String, Boolean>? = null,
    val allowedModels: List<String>? = null,
)

@Serializable
data class PutAiConfigRequest(
    val featuresEnabled: Map<String, Boolean>,
    val allowedModels: List<String>? = null,
)

@Serializable
data class AiProviderSettingsResponse(
    val orgId: String? = null,
    val provider: String? = null,
    val modelAlias: String? = null,
    val fallbackProvider: String? = null,
    val byokConfigured: Boolean? = null,
    val settings: Map<String, JsonElement>? = null,
    val providers: List<String>? = null,
    val modelAliases: List<String>? = null,
)

@Serializable
data class PutAiProviderSettingsRequest(
    val provider: String,
    val modelAlias: String,
    val fallbackProvider: String? = null,
    @SerialName("byokApiKey") val byokApiKey: String? = null,
)

@Serializable
data class AiProviderTestResponse(
    val provider: String? = null,
    val latencyMs: Int? = null,
    val responsePreview: String? = null,
)

File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsOrgBrandingAdmin.swift
import Foundation

// MARK: - Org branding / AI governance / AI provider admin (M14.5)

struct OrgBrandingResponse: Decodable, Hashable {
    var logoUrl: String?
    var faviconUrl: String?
    var primaryColor: String
    var secondaryColor: String
    var customDomain: String?
    var customEmailDisplayName: String?
    var contrastWarningPrimary: Bool?
    var contrastRatioPrimary: Double?

    init(
        logoUrl: String? = nil,
        faviconUrl: String? = nil,
        primaryColor: String = OrgBrandingAdminLogic.defaultPrimaryColor,
        secondaryColor: String = OrgBrandingAdminLogic.defaultSecondaryColor,
        customDomain: String? = nil,
        customEmailDisplayName: String? = nil,
        contrastWarningPrimary: Bool? = nil,
        contrastRatioPrimary: Double? = nil
    ) {
        self.logoUrl = logoUrl
        self.faviconUrl = faviconUrl
        self.primaryColor = primaryColor
        self.secondaryColor = secondaryColor
        self.customDomain = customDomain
        self.customEmailDisplayName = customEmailDisplayName
        self.contrastWarningPrimary = contrastWarningPrimary
        self.contrastRatioPrimary = contrastRatioPrimary
    }

    enum CodingKeys: String, CodingKey {
        case logoUrl, faviconUrl, primaryColor, secondaryColor
        case customDomain, customEmailDisplayName
        case contrastWarningPrimary, contrastRatioPrimary
    }

    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        logoUrl = try c.decodeIfPresent(String.self, forKey: .logoUrl)
        faviconUrl = try c.decodeIfPresent(String.self, forKey: .faviconUrl)
        primaryColor = try c.decodeIfPresent(String.self, forKey: .primaryColor)
            ?? OrgBrandingAdminLogic.defaultPrimaryColor
        secondaryColor = try c.decodeIfPresent(String.self, forKey: .secondaryColor)
            ?? OrgBrandingAdminLogic.defaultSecondaryColor
        customDomain = try c.decodeIfPresent(String.self, forKey: .customDomain)
        customEmailDisplayName = try c.decodeIfPresent(String.self, forKey: .customEmailDisplayName)
        contrastWarningPrimary = try c.decodeIfPresent(Bool.self, forKey: .contrastWarningPrimary)
        contrastRatioPrimary = try c.decodeIfPresent(Double.self, forKey: .contrastRatioPrimary)
    }
}

struct OrgBrandingPutRequest: Encodable {
    var logoUrl: String?
    var faviconUrl: String?
    var primaryColor: String
    var secondaryColor: String
    var customDomain: String?
    var customEmailDisplayName: String?
}

struct OrgBrandingUploadResponse: Decodable {
    var url: String?
}

struct AIGovernanceConfig: Decodable, Hashable {
    var orgId: String?
    var featuresEnabled: [String: Bool]?
    var allowedModels: [String]?
    var updatedAt: String?
    var updatedBy: String?
}

struct AIGovernancePutRequest: Encodable {
    var featuresEnabled: [String: Bool]
    var allowedModels: [String]?
}

struct AIProviderSettings: Decodable, Hashable {
    var orgId: String?
    var provider: String?
    var modelAlias: String?
    var fallbackProvider: String?
    var byokConfigured: Bool?
    var providers: [String]?
    var modelAliases: [String]?
    var updatedAt: String?
    var updatedBy: String?
}

struct AIProviderSettingsPutRequest: Encodable {
    var provider: String
    var modelAlias: String
    var fallbackProvider: String?
    var byokApiKey: String?

    enum CodingKeys: String, CodingKey {
        case provider, modelAlias, fallbackProvider, byokApiKey
    }

    func encode(to encoder: Encoder) throws {
        var c = encoder.container(keyedBy: CodingKeys.self)
        try c.encode(provider, forKey: .provider)
        try c.encode(modelAlias, forKey: .modelAlias)
        try c.encodeIfPresent(fallbackProvider, forKey: .fallbackProvider)
        try c.encodeIfPresent(byokApiKey, forKey: .byokApiKey)
    }
}

struct AIProviderTestResponse: Decodable {
    var provider: String?
    var latencyMs: Double?
    var responsePreview: String?
}
